src/libAtomVM/otp_socket.c

```diff
@@ -1323,6 +1328,52 @@ static term nif_socket_setopt(Context *ctx, int argc, term argv[])
                 }
             }
 
+#if OTP_SOCKET_BSD
+            case OtpSocketSetoptLevelIP: {
+                term opt = term_get_tuple_element(level_tuple, 1);
+                if (globalcontext_is_term_equal_to_atom_string(global, opt, add_membership_atom)) {
+                    // socket:setopt(Socket, {ip, add_membership_atom}, Req :: ip_mreq())
+
+                    if (UNLIKELY(!term_is_map(value))) {
+                        TRACE("socket:setopt: ip add_membership_atom value must be a map");
+                        SMP_RWLOCK_UNLOCK(rsrc_obj->socket_lock);
+                        return make_error_tuple(globalcontext_make_atom(global, invalid_value_atom), ctx);
+                    }
+
+                    term multiaddr = interop_map_get_value(ctx->global, value, globalcontext_make_atom(global, multiaddr_atom));
```

Comment:
interop_kv_get_value(_default) might be a better alternative: it doesn't require creating an atom in advance.
e.g. `interop_kv_get_value(value, ATOM_STR("\x9", "multiaddr"), ctx->global)`

Reply:
Also I suggest making lines shorter by keeping a pointer to GlobalContext so we don't end writing ctx->global at every line.